components/org.wso2.carbon.identity.sso.saml/src/test/java/org/wso2/carbon/identity/sso/saml/util/SAMLSSOUtilTest.java

```diff
@@ -18,18 +18,206 @@
+import static org.powermock.api.mockito.PowerMockito.mockStatic;
+import static org.powermock.api.mockito.PowerMockito.when;
+import static org.testng.Assert.*;
+
+/**
+ * Unit test cases for SAMLSSOUtil.
+ */
+@PrepareForTest(IdentityProviderManager.class)
 public class SAMLSSOUtilTest {
-    @org.testng.annotations.Test
+
+    @Mock
+    private RealmService realmService;
+
+    @Mock
+    private TenantManager tenantManager;
+
+    @Mock
+    private IdentityProviderManager identityProviderManager;
+
+    @Mock
+    private IdentityProvider identityProvider;
+
+    @Mock
+    private FederatedAuthenticatorConfig federatedAuthenticatorConfig;
+
+    @ObjectFactory
+    public IObjectFactory getObjectFactory() {
+        return new org.powermock.modules.testng.PowerMockObjectFactory();
+    }
+
+    @BeforeTest
+    public void initTest() throws UserStoreException, IdentityProviderManagementException {
+        MockitoAnnotations.initMocks(this);
+    }
+
+    private void prepareForGetIssuer() throws UserStoreException,
+            IdentityProviderManagementException {
+
+        when(tenantManager.getTenantId(anyString())).thenReturn(-1234);
+        when(realmService.getTenantManager()).thenReturn(tenantManager);
+
+        SAMLSSOUtil.setRealmService(realmService);
+
+        Property property = new Property();
+        property.setName("IdPEntityId");
+        property.setValue(TestConstants.LOACALHOST_DOMAIN);
+        Property[] properties = {property};
+        when(federatedAuthenticatorConfig.getProperties()).thenReturn(properties);
+        when(federatedAuthenticatorConfig.getName()).thenReturn(
+                IdentityApplicationConstants.Authenticator.SAML2SSO.NAME);
+        FederatedAuthenticatorConfig[] fedAuthConfs = {federatedAuthenticatorConfig};
+        when(identityProvider.getFederatedAuthenticatorConfigs()).thenReturn(fedAuthConfs);
+
+        mockStatic(IdentityProviderManager.class);
+        when(IdentityProviderManager.getInstance()).thenReturn(identityProviderManager);
+        when(identityProviderManager.getResidentIdP(anyString())).thenReturn(identityProvider);
+    }
+
+    @Test
+    public void testGetIssuer() throws Exception {
+
+        prepareForGetIssuer();
+        Issuer issuer = SAMLSSOUtil.getIssuer();
+        assertEquals(issuer.getValue(), TestConstants.LOACALHOST_DOMAIN, "Super tenant mode.");
+        assertEquals(issuer.getFormat(), SAMLSSOConstants.NAME_ID_POLICY_ENTITY,
+                "Issuer format should always be SAML2 spec compatible");
+    }
+
+    @Test
+    public void testGetIssuerFromTenantDomain() throws Exception {
+
+        prepareForGetIssuer();
+        Issuer issuer = SAMLSSOUtil.getIssuerFromTenantDomain(MultitenantConstants.SUPER_TENANT_DOMAIN_NAME);
+        assertEquals(issuer.getValue(), TestConstants.LOACALHOST_DOMAIN, "Issuer for uuper tenant domain.");
+    }
+
+    @Test
+    public void testGetDestinationFromTenantDomain() throws Exception {
```

Comment:
Shall we remove these unimplemented tests for the time being ? Iet's add them when you actually implement them

Reply:
Since these tests will be implemented in a short while it is not a big deal right? 
These are the identified test cases that need to be implemented for this class.

Reply:
Shot while is subjective :). In a case if we fail to implement these tests due to some reason, these tests will also count in reports. So it's misleading.